composer/benchmarker/benchmarker.py

```diff
@@ -0,0 +1,110 @@
+
+from composer.benchmarker.benchmarker_hparams import BenchmarkerHparams
+from composer.callbacks.benchmarker import Benchmarker as BenchmarkerCallback
+from composer.core.logging.base_backend import BaseLoggerBackend
+from composer.core.precision import Precision
+from composer.datasets.hparams import DataloaderSpec
+from composer.datasets.synthetic import SyntheticDataLabelType, SyntheticDataset
+from composer.models.base import BaseMosaicModel
+from composer.trainer.devices import DeviceCPU, DeviceGPU
+from composer.trainer.trainer import Trainer
+
+NUM_PROFILING_STEPS = 250  # needs to be high enough to be able to properly measure time
+
+
+class Benchmarker:
+    """Benchmarker for timing model training workloads with synthetic data.
+
+    Args:
+        model (BaseMosaicModel): The model to profile.
+        data_shape (List[int]): Shape of the tensor for input samples.
+        total_batch_size (int): The batch size to train with.
+        grad_accum (int, optional): The number of microbatches to split a per-device batch into. Gradients
+            are summed over the microbatches per device.
+        label_type (SyntheticDataLabelType, optional), Type of synthetic data to create.
+            If `CLASSIFICATION_INT` or `CLASSIFICATION_ONE_HOT` then `num_classes` must be specified.
+            If `RANDOM_INT` then `label_shape` must be specified.
+        num_classes (int, optional): Number of classes to use.
+        label_shape (List[int]): Shape of the tensor for each sample label.
+        log_destinations (List[BaseLoggerBackend], optional): The destinations to log training information to.
+    """
+
+    def __init__(self,
+                 model: BaseMosaicModel,
+                 data_shape: Sequence[int],
+                 total_batch_size: int,
+                 grad_accum: int,
+                 label_type: SyntheticDataLabelType = SyntheticDataLabelType.CLASSIFICATION_INT,
+                 num_classes: Optional[int] = None,
+                 label_shape: Optional[Sequence[int]] = None,
+                 log_destinations: Optional[List[BaseLoggerBackend]] = None):
+
+        dataset_size = total_batch_size * NUM_PROFILING_STEPS
+        self.dataset = SyntheticDataset(total_dataset_size=dataset_size,
+                                        data_shape=data_shape,
+                                        num_unique_samples_to_create=total_batch_size,
+                                        label_type=label_type,
+                                        num_classes=num_classes,
+                                        label_shape=label_shape)
+
+        self.dataloader_spec = DataloaderSpec(dataset=self.dataset, shuffle=False, drop_last=True)
+
+        # Default for now - adjust to work with algorithms
+        timing_callback = BenchmarkerCallback(min_steps=50, epoch_list=[0, 1], step_list=[0, 50], all_epochs=False)
+
+        # Use optimal device settings based on what is available
+        device = DeviceCPU()
```

Comment:
can you add the device in as an optional argument? I think this assumption is fine if it's not specified, but during tests, it would be better to be explicit